src/chess/piece/Pawn.java

```diff
@@ -12,16 +12,37 @@
         }
       };
 
+
   public Pawn(Color color) {
     super(color, symbols);
   }
 
   @Override
-  public boolean isValidMove(Position newPosition) {
-    if (!super.isValidMove(newPosition)) {
+  public boolean isValidMove(int x, int y, boolean isEnemyExisted) {
+    if (!super.isValidMove(x, y, isEnemyExisted)) {
       return false;
     }
-    // ToDo: implement
-    return true;
+
+    // this piece can move forward
+    if (this.color == Color.BLACK) {
+      x = x * -1;
+      y = y * -1;
+    }
+
+    // x = -1, 0, 1
+    // y = 1, 2
+    if (x < -1 || 1 < x) return false;
+    if (y < 0 || 2 < y) return false;
+
+    // Diagonally forward
+    if (isEnemyExisted && y == 1 && Math.abs(x) == 1) return true;
+
+    // forward two
+    if (!isEnemyExisted && lastMovedTurn == 0 && x == 0 && y == 2) return true;
+
+    // forward one
+    if (!isEnemyExisted && x == 0 && y == 1) return true;
```

Comment:
This calculation should be done as `Board` responsibility, but acceptable for now:+1:

Reply:
I didn't improve it. Please polish it 🙇